axonserver/src/test/java/io/axoniq/axonserver/message/command/CommandDispatcherTest.java

```diff
@@ -322,4 +326,48 @@ public SerializedCommandResponse commandResponse(
         SerializedCommandResponse response = futureResponse.get();
         assertEquals(ErrorCode.EXCEPTION_IN_INTERCEPTOR.getCode(), response.getErrorCode());
     }
+
+    @Test
+    public void duplicateCommandGetsRejected() throws ExecutionException, InterruptedException {
+        // See QueryDispatcherTest.queryDuplicated for explanation
+        String duplicatedId = "duplicatedId";
+
+        commandCache = new CommandCache(10000, Clock.systemUTC(), 100000);
+
+        CompletableFuture<SerializedCommandResponse> originalFutureResponse = new CompletableFuture<>();
+        CommandInformation commandInformation = new CommandInformation(duplicatedId,"","",
+                                                                       originalFutureResponse::complete,
+                                                                       new ClientStreamIdentification(Topology.DEFAULT_CONTEXT,"client"),"component");
+
+        commandCache.put(duplicatedId,commandInformation);
+
+        commandDispatcher = new CommandDispatcher(registrations, commandCache, metricsRegistry, meterFactory,new NoOpCommandInterceptors(), 10_000);
+
+        FakeStreamObserver<SerializedCommandProviderInbound> commandProviderInbound = new FakeStreamObserver<>();
+        ClientStreamIdentification client = new ClientStreamIdentification(Topology.DEFAULT_CONTEXT, "client");
+        DirectCommandHandler result = new DirectCommandHandler(commandProviderInbound,
+                                                               client, "client", "component");
+        when(registrations.getHandlerForCommand(eq(Topology.DEFAULT_CONTEXT), any(), any())).thenReturn(result);
+
+        CompletableFuture<SerializedCommandResponse> futureResponse = new CompletableFuture<>();
+        commandDispatcher.dispatch(Topology.DEFAULT_CONTEXT, null,
+                                   new SerializedCommand(Command.newBuilder()
+                                                                .setMessageIdentifier(duplicatedId)
+                                                                .build()), futureResponse::complete);
+
+
+        commandDispatcher.handleResponse(new SerializedCommandResponse(CommandResponse.newBuilder()
+                                                                                      .setRequestIdentifier(duplicatedId)
+                                                                                      .build()), false);
+
+        try {
+            SerializedCommandResponse originalResponse = originalFutureResponse.get(1, TimeUnit.SECONDS);
+            assertEquals(duplicatedId, originalResponse.getRequestIdentifier());
+        } catch (TimeoutException e) {
+            fail();
```

Comment:
I don't believe there is need for this: if the `TimeoutException` is thrown, the test will fail anyway :)